<a href="https://colab.research.google.com/github/JulTob/Modelling/blob/main/Muestreo/T2/E2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Ejercicio
En una población adulta de $N = 10_. 000$ individuos se efectúa un muestreo aleatorio simple de tamaño $n$ para determinar su opinión respecto a un cierto tema.

Se obtienen los siguientes resultados:

Grupo | A favor | En contra | Indecisos | Total
------|---|---|---|---
Total muestral | 90 | 80 | 30 | 200





<font color="goldenrod">
1. Extracción de datos

$$N = 10000$$
$$n=200$$

Grupo | A favor | En contra | Indecisos | Total
------|---|---|---|---
Total muestral | 90 | 80 | 30 | 200
Frecuencia | 0.45 | 0.4 | 0.15 | 1

In [19]:
import pandas as pd

N_Poblacion = 10_000
n_muestra = 200

data = pd.DataFrame({
    'Grupo': ['Total muestral', 'Frecuencia muestral'],
    'A favor': [90,             90/n_muestra],
    'En contra': [80,           80/n_muestra],
    'Indecisos': [30,           30/n_muestra],
    'Total': [200,              1]
    })

data

,Grupo,A favor,En contra,Indecisos,Total
0,Total muestral,90.00,80.0,30.00,200
1,Frecuencia muestral,0.45,0.4,0.15,1


a) Calcular los límites de confianza al `67 %` para el estimador de la proporción de
los que están a favor.

In [25]:
Total = 0
Frecuencia = 1

p̂ = data['A favor'][Frecuencia]
p̂

np.float64(0.45)

In [21]:
# Error estándar de la proporción muestral
# Varianza de distr. binomial: p̂ * ¬p̂
SE = np.sqrt(p̂ * (1 - p̂) / n)

SE

np.float64(0.03517811819867572)

b) ¿Es el porcentaje de los que están a favor significativamente más pequeño que
el `50 %` con un nivel de confianza del `95 %`?

c) ¿Qué tamaño muestral debe tomarse para no cometer en la estimación un error
mayor de `±2` puntos porcentuales con un nivel de confianza del `95 %`?

**Reasoning**:
The subtask requires extracting the given data and calculating the sample proportion of those 'A favor'. I will assign the population size (N), sample size (n), and the number of individuals 'A favor' to variables, then calculate the sample proportion (p_hat).



**Reasoning**:
To calculate the 67% confidence interval, I need to import the `norm` function from `scipy.stats`, calculate the standard error, determine the critical Z-value, compute the margin of error, and finally calculate the lower and upper bounds of the confidence interval using the provided formulas and print all relevant values.



In [22]:
import numpy as np
from scipy.stats import norm



# 3. Determinar el valor crítico Z para un nivel de confianza del 67%
confidence_level = 0.67
alpha = 1 - confidence_level
Z_critical = norm.ppf(1 - alpha / 2)

# 4. Calcular el margen de error
margin_of_error = Z_critical * SE

# 5. Calcular los límites inferior y superior del intervalo de confianza
lower_bound = p_hat - margin_of_error
upper_bound = p_hat + margin_of_error

# 6. Imprimir los resultados
print(f"Error Estándar (SE): {SE:.4f}")
print(f"Valor Crítico Z para 67% de confianza: {Z_critical:.4f}")
print(f"Margen de Error: {margin_of_error:.4f}")
print(f"Intervalo de Confianza al 67%: [{lower_bound:.4f}, {upper_bound:.4f}]")

Error Estándar (SE): 0.0352
Valor Crítico Z para 67% de confianza: 0.9741
Margen de Error: 0.0343
Intervalo de Confianza al 67%: [0.4157, 0.4843]


## Realizar Prueba de Hipótesis (Parte b)

### Subtask:
Determinar si el porcentaje de los que están 'A favor' es significativamente más pequeño que 50% con un nivel de confianza del 95%.


### Reasoning for Hypothesis Test (Part b):
To determine if the percentage of those 'A favor' is significantly smaller than 50% with a 95% confidence level, we will perform a one-tailed hypothesis test for proportions.

1.  **Formulate Hypotheses**:
    *   **Hypothesis nula (H0)**: La proporción de los que están a favor es igual o mayor al 50% (p >= 0.5).
    *   **Hypothesis alternativa (Ha)**: La proporción de los que están a favor es menor al 50% (p < 0.5).

2.  **Nivel de Significancia (alpha)**:
    *   Para un nivel de confianza del 95%, el nivel de significancia para una prueba unilateral es `alpha = 0.05`.

3.  **Calcular el Error Estándar bajo H0 (SE_0)**:
    *   Cuando se prueba una hipótesis sobre una proporción, el error estándar se calcula usando la proporción hipotética (p0) de la hipótesis nula. Aquí, p0 = 0.5.
    *   Fórmula: `SE_0 = sqrt(p0 * (1 - p0) / n)`

4.  **Calcular el Estadístico de Prueba Z**:
    *   Fórmula: `Z = (p_hat - p0) / SE_0`

5.  **Determinar el Valor Crítico Z**:
    *   Para una prueba unilateral izquierda con `alpha = 0.05`, encontramos el valor Z tal que el área a su izquierda es 0.05. Este valor se obtiene usando `norm.ppf(alpha)`.

6.  **Conclusión**:
    *   Si el estadístico de prueba Z es menor que el valor crítico Z, rechazamos la hipótesis nula (H0). De lo contrario, no la rechazamos.

**Reasoning**:
Now, I will execute the calculations for the hypothesis test in a code block, following the steps outlined in the previous markdown. This includes calculating the standard error under the null hypothesis, the Z-statistic, the critical Z-value, and finally determining whether to reject the null hypothesis.



In [23]:
import numpy as np
from scipy.stats import norm

# Variables from previous steps:
# N = 10000
# n = 200
# p_hat = 0.45

# 1. Hypotheses (already stated in markdown)
# H0: p >= 0.5
# Ha: p < 0.5

# 2. Nivel de significancia (alpha)
alpha_ht = 0.05  # For 95% confidence level, one-tailed test

# 3. Proporción hipotética bajo H0
p0 = 0.5

# 4. Calcular el error estándar bajo H0 (SE_0)
SE_0 = np.sqrt(p0 * (1 - p0) / n)

# 5. Calcular el estadístico de prueba Z
Z_statistic = (p_hat - p0) / SE_0

# 6. Determinar el valor crítico Z para una prueba unilateral izquierda
Z_critical_ht = norm.ppf(alpha_ht)

# 7. Comparar el estadístico de prueba Z con el valor crítico Z y concluir
print(f"Proporción muestral (p_hat): {p_hat}")
print(f"Proporción hipotética (p0): {p0}")
print(f"Nivel de significancia (alpha): {alpha_ht}")
print(f"Error Estándar bajo H0 (SE_0): {SE_0:.4f}")
print(f"Estadístico de Prueba Z: {Z_statistic:.4f}")
print(f"Valor Crítico Z para alpha={alpha_ht} (unilateral izquierda): {Z_critical_ht:.4f}")

if Z_statistic < Z_critical_ht:
    print("Conclusión: Se rechaza la hipótesis nula (H0). Hay evidencia de que la proporción de los que están a favor es significativamente menor al 50%.")
else:
    print("Conclusión: No se rechaza la hipótesis nula (H0). No hay evidencia suficiente para afirmar que la proporción de los que están a favor es significativamente menor al 50%.")

Proporción muestral (p_hat): 0.45
Proporción hipotética (p0): 0.5
Nivel de significancia (alpha): 0.05
Error Estándar bajo H0 (SE_0): 0.0354
Estadístico de Prueba Z: -1.4142
Valor Crítico Z para alpha=0.05 (unilateral izquierda): -1.6449
Conclusión: No se rechaza la hipótesis nula (H0). No hay evidencia suficiente para afirmar que la proporción de los que están a favor es significativamente menor al 50%.


## Calcular Tamaño Muestral Requerido (Parte c)

### Subtask:
Determinar el tamaño de muestra necesario para no cometer un error mayor de ±2 puntos porcentuales con un nivel de confianza del 95%.


### Reasoning for Sample Size Calculation (Part c):
To determine the necessary sample size, we will follow these steps:

1.  **Define the desired confidence level (95%)** and calculate the corresponding critical Z-value. For a 95% confidence, `alpha` is 0.05, and we need the Z-value for `1 - alpha/2` (i.e., `norm.ppf(0.975)`).
2.  **Define the maximum allowable margin of error (E)**. The problem specifies \u00b12 percentage points, which translates to E = 0.02 in decimal form.
3.  **Establish an estimated proportion (p)**. Since no prior estimate is available, we will use p = 0.5. This value maximizes the term `p * (1 - p)`, leading to the largest and most conservative (safe) sample size.
4.  **Apply the formula for the sample size** of a proportion:
    $$ n = \frac{Z^2 \cdot p \cdot (1-p)}{E^2} $$
5.  **Round the result up** to the nearest whole number, as the sample size must be an integer, and rounding up ensures that the error does not exceed the specified margin.
6.  Print the calculated sample size and all important intermediate values (Z, E, p).

**Reasoning**:
Now, I will implement the sample size calculation as described in the previous markdown block. This involves defining the confidence level, margin of error, and estimated proportion, then using the `norm.ppf` function to find the critical Z-value, applying the sample size formula, and finally rounding up the result to get the required sample size.



In [24]:
import numpy as np
from scipy.stats import norm
import math

# 1. Nivel de confianza deseado y cálculo del valor crítico Z
confidence_level_c = 0.95
alpha_c = 1 - confidence_level_c
Z_critical_c = norm.ppf(1 - alpha_c / 2)

# 2. Margen de error máximo permitido (E)
E = 0.02  # 2 puntos porcentuales en formato decimal

# 3. Proporción estimada (p)
p = 0.5  # Usar 0.5 para el tamaño de muestra más conservador

# 4. Aplicar la fórmula para el tamaño de la muestra
n_calculated = (Z_critical_c**2 * p * (1 - p)) / (E**2)

# 5. Redondear el resultado al entero superior más cercano
n_required = math.ceil(n_calculated)

# 6. Imprimir los resultados
print(f"Nivel de confianza: {confidence_level_c * 100}%")
print(f"Valor Crítico Z: {Z_critical_c:.4f}")
print(f"Margen de Error (E): {E}")
print(f"Proporción estimada (p): {p}")
print(f"Tamaño de muestra calculado (sin redondear): {n_calculated:.2f}")
print(f"Tamaño de muestra requerido: {n_required}")

Nivel de confianza: 95.0%
Valor Crítico Z: 1.9600
Margen de Error (E): 0.02
Proporción estimada (p): 0.5
Tamaño de muestra calculado (sin redondear): 2400.91
Tamaño de muestra requerido: 2401


## Final Task

### Subtask:
Resumir los resultados y las conclusiones obtenidas de cada parte del ejercicio (a, b y c).


## Summary:

### Q&A
1.  **What is the 67% confidence interval for the proportion of those 'A favor'?**
    The 67% confidence interval for the proportion of those 'A favor' is \[0.4157, 0.4843]. This means we are 67% confident that the true proportion of individuals 'A favor' in the population lies between 41.57% and 48.43%.

2.  **Is the percentage of those 'A favor' significantly smaller than 50% with a 95% confidence level?**
    No, at a 95% confidence level, there is not sufficient evidence to conclude that the proportion of individuals 'A favor' is significantly smaller than 50%. The hypothesis test yielded a Z-statistic of -1.4142, which is not less than the critical Z-value of -1.6449 for a one-tailed test.

3.  **What is the required sample size to estimate the proportion with an error not greater than \$\pm\$2 percentage points with a 95% confidence level?**
    The required sample size is 2401 individuals to estimate the proportion with a margin of error not exceeding \$\pm\$2 percentage points at a 95% confidence level, assuming a conservative estimated proportion of 0.5.

### Data Analysis Key Findings
*   From a sample of 200 individuals, 90 were 'A favor', resulting in a sample proportion (\$\hat{p}\$) of 0.45 (45%).
*   The 67% confidence interval for the proportion of those 'A favor' is \[0.4157, 0.4843], with a margin of error of 0.0343.
*   A one-tailed hypothesis test at a 95% confidence level (significance level \$\alpha\$ = 0.05) to determine if the proportion 'A favor' is significantly less than 50% resulted in a Z-statistic of -1.4142.
*   Since the Z-statistic (-1.4142) is greater than the critical Z-value (-1.6449), the null hypothesis (proportion \$\ge\$ 0.5) cannot be rejected.
*   To estimate the population proportion with a margin of error of \$\pm\$2 percentage points and a 95% confidence level, a sample size of 2401 individuals is required.

### Insights or Next Steps
*   Although the sample proportion of 'A favor' (45%) is less than 50%, this difference is not statistically significant at the 95% confidence level. Further investigation might be needed if a smaller difference is considered important.
*   The current sample size of 200 is insufficient for future estimations if a margin of error of \$\pm\$2% with 95% confidence is desired, as a much larger sample size of 2401 would be needed.
